pkg/integrations/v2/app_o11y_receiver/exporters/logs.go

```diff
@@ -0,0 +1,128 @@
+package exporters
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	kitlog "github.com/go-kit/log"
+	"github.com/go-kit/log/level"
+	"github.com/go-logfmt/logfmt"
+	"github.com/grafana/agent/pkg/integrations/v2/app_o11y_receiver/models"
+	"github.com/grafana/agent/pkg/integrations/v2/app_o11y_receiver/sourcemaps"
+	"github.com/grafana/agent/pkg/integrations/v2/app_o11y_receiver/utils"
+	"github.com/grafana/agent/pkg/logs"
+	"github.com/grafana/loki/clients/pkg/promtail/api"
+	"github.com/grafana/loki/pkg/logproto"
+	prommodel "github.com/prometheus/common/model"
+)
+
+type logsInstance interface {
+	SendEntry(entry api.Entry, dur time.Duration) bool
+}
+
+// LogsExporterConfig holds the configuration of the logs exporter
+type LogsExporterConfig struct {
+	SendEntryTimeout int
+	LogsInstance     logsInstance
```

Comment:
nit: the type of an exported field on an exported struct should also be exported (i.e., export `logsInstance`)